src/solve.jl

```diff
@@ -0,0 +1,84 @@
+
+function DiffEqBase.solve(prob::OptimizationProblem, opt, args...;kwargs...)
+	__solve(prob, opt, args...; kwargs...)
+end
+
+decompose_trace(trace::Optim.OptimizationTrace) = last(trace)
+decompose_trace(trace) = trace
+
+function __solve(prob::OptimizationProblem, opt::Optim.ZerothOrderOptimizer;cb = (args...) -> (false), kwargs...)
```

Comment:
no reason to separate this out: these 3 dispatches are essentially the same and this is just duplicating code